chall03/mpena-gu.py

```diff
@@ -58,27 +62,35 @@ def snail (row_init, row_end, col_init, col_end, squared_rows):
     snail_str = ", ".join(snail_result)
     return snail_str
 
-if __name__ == "__main__":
-    if len(sys.argv) < 2 or not check_argv(sys.argv[1:]):
-        print ("usage: ./mpena-gu.py <1-9 squared_rows...>")
-        sys.exit()
-    size_array = len(sys.argv[1])
-    if size_array == 1:
+def resolve():
+    if len(sys.argv[1:]) == 1:
+        print("%s" % sys.argv[1])
+    elif len(sys.argv[1:]) == 2 and len(sys.argv[1]) == 1:
         print("%s, %s" % (sys.argv[1],sys.argv[2]))
+    else:
+        size_array = len(sys.argv[1])
+        squared = create_squared_rows(sys.argv[1:])
+        num_squared = size_array//2 
+        if size_array%2 != 0:
+            num_squared += 1
+        i = 0 
+        snail_result = "" 
+        while i< num_squared:
+            if i>0:
+                snail_result += ', '
+            row_init = i
+            row_end  = size_array - i
+            col_init = i
+            col_end  = size_array - i
+            snail_result += snail(row_init,row_end,col_init,col_end, squared)
+            i += 1
+        print(snail_result)
```

Comment:
You had already sanitised the input so, you could have printed the solution character by character, instead of wasting memory storing everything inside a variable and printing it at the end.